libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::abs;
+use crate::bignum::utils::recast_vec;
+use crate::bignum::Bignum;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        let mut is_neg: bool = false;
+        let lhs: Bignum;
+        let rhs: Bignum;
+        if !self.is_neg && other.is_neg {
+            return self + abs(&other);
+        } else if self.is_neg && !other.is_neg {
+            return self + (-other);
+        }
+        // make rhs the bigger number
+        if abs(&self) > abs(&other) {
+            if self.is_neg {
+                is_neg = true;
+            }
+            lhs = other;
+            rhs = self;
+        } else {
+            // this really is !self.is_neg && !other.is_neg but we can use demorgans for speed
+            if !(self.is_neg || other.is_neg) {
+                is_neg = true;
+            }
```

Comment:
It would be even faster if you set this on line 11 and avoided the branch.

Reply:
you can't do this, I am already avoiding branches by setting it to false on line 11

Reply:
Why can't you do it? Just do is_neg = self.is_neg || other.is_neg.

Reply:
oh i see ok.

Reply:
I was thinking  a different idea

Reply:
this idea doesn't work. It needs to be a specific case where the lhs is < rhs to work. I think the way I have it is fine.

Reply:
How can that be the case? If they're both negative numbers the result must be negative.

Reply:
no, this is handling if  they are both positive and lhs is < rhs.

Reply:
got it